app/controllers/katello/api/v2/host_module_streams_controller.rb

```diff
@@ -38,6 +38,14 @@ def resource_class
 
     private
 
+    def find_available_module_stream_ids
+      items = ::Katello::AvailableModuleStream.select(:name, :stream, :id)
+      grouped = items.group_by { |item| [item.name, item.stream] }
```

Comment:
If we see performance problems in the future with this query, it's probably because of the `group_by` here which uses Ruby and not SQL to organize the data. There's probably a way to do it with Active Record instead (`.group`), but I'd say let's not prematurely optimize. So I'm fine with this for now :)

Reply:
@jeremylenz Good point but I don't think we have to worry about that. We are grouping module streams available for the host.  For my test centos 8 host there are 459 module streams.
I tried the following test
```ruby
def time_it
  start = (Time.now.to_f * 1000).to_i
  yield
  (Time.now.to_f * 1000).to_i - start
end  

>  x = [5] * 10000;x.count
=> 10000

time_it { x.group_by {|x|x} }
=> 3
```
 So it takes 3 milliseconds to group 10K items. Dont think we have to worry about this.

That being said lets ensure that we are only grouping available module streams for one host and not all.

@chris1984 would recommend
```ruby
    def find_available_module_stream_ids
      items = @host.host_available_module_streams.includes(:available_module_stream).map(&:available_module_stream) 
      grouped = items.group_by { |item| [item.name, item.stream] }
      grouped.values.map do |item|
        item.first.id
      end
    end
```